pkg/skaffold/util/store.go

```diff
@@ -17,33 +17,67 @@ limitations under the License.
 package util
 
 import (
+	"fmt"
 	"sync"
+
+	"github.com/golang/groupcache/singleflight"
 )
 
-// SyncStore exports a single method `Exec` to ensure single execution of a function and share the result between all callers of the function.
+// SyncStore exports a single method `Exec` to ensure single execution of a function
+// and share the result between all callers of the function.
 type SyncStore struct {
-	oncePerKey *sync.Map
-	results    *sync.Map
+	prefix  string
+	sf      singleflight.Group
```

Comment:
shouldn't `sf` be `*singleflight.Group`? Otherwise won't the underlying `Mutex` get copied on every reference of the instantiated struct?

Reply:
Not really, i usually refer the tests as a documentation for using code.

https://github.com/golang/sync/blob/master/singleflight/singleflight_test.go#L23

Here i also realized we don't need sync.Map as pointer. 
https://play.golang.org/p/iramynV7eJV